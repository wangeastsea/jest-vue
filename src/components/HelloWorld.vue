<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <p>123</p>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  props: {
    msg: String
  }
}
</script>
<style scoped lang="stylus">
</style>
