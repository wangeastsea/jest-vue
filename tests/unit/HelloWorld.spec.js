import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message'
    // shallowMount不渲染子组件，提供性能，不适用于集成测试
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    })
    expect(wrapper.text()).toMatch(msg)
    expect(wrapper.props('msg')).toEqual(msg)
    wrapper.setProps({
      msg: 'hello'
    })
    expect(wrapper.props('msg')).toEqual('hello')
  })
  // 快照测试
  it('组件渲染正常', () => {
    const wrapper = shallowMount(HelloWorld, {
      propsData: {
        msg: 'de2ll lee'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
