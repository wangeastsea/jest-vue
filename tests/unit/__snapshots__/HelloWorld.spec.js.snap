// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue 组件渲染正常 1`] = `
<div class="hello">
  <h1>de2ll lee</h1>
  <p>123</p>
</div>
`;
